### PR TITLE
Bump DB2 Driver to 12.1.0.0

### PR DESCRIPTION
**Pull request by gastaldi**
It also uses the `jakarta.transaction` packages, so no need to transform it

### CODE REVIEW

**Comment by gastaldi** [comment]
> You were able to compile to native? The Dependabot update was complaining?

I was waiting for CI to do that because there is no support for DB2 DevService in `arm64`: 

> [ERROR] DOCKER> I/O Error [Unable to pull 'icr.io/db2_community/db2:11.5.9.0' from registry 'icr.io' : {"message":"no image found in manifest list for architecture \"arm64\", variant \"v8\", OS \"linux\""} (Internal Server Error: 500)]

**Comment by gsmet** [comment]
You will probably end up with the error in https://github.com/quarkusio/quarkus/pull/44665 (see my comment).

**Comment by gastaldi** [comment]
@gsmet it passes now

File: extensions/jdbc/jdbc-db2/runtime/src/main/java/io/quarkus/jdbc/db2/runtime/graal/ConnectionSubstitution.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(Connection.class)
+public final class ConnectionSubstitution {
+
+    @Substitute
+    private void checkForLicenseRestrictions() throws SQLException {
+        // Do nothing
+    }
```
Comment (gastaldi) [inline extensions/jdbc/jdbc-db2/runtime/src/main/java/io/quarkus/jdbc/db2/runtime/graal/ConnectionSubstitution.java:15]: Not really proud of it, but the method body is uncompilable and unknown

**Comment by gastaldi** [comment]
@gsmet I found a nicer solution

**Comment by gastaldi** [comment]
Now it fails with: 

```java
Caused by: java.lang.UnsupportedOperationException: 'TCP_KEEPIDLE' not supported
	at java.base@21.0.5/sun.nio.ch.NioSocketImpl.setOption(NioSocketImpl.java:959)
	at java.base@21.0.5/java.net.DelegatingSocketImpl.setOption(DelegatingSocketImpl.java:144)
	at java.base@21.0.5/java.net.Socket.setOption(Socket.java:2009)
	at com.ibm.db2.jcc.t4.w.a(w.java:76)
	at com.ibm.db2.jcc.t4.az.a(az.java:441)
	... 21 more
```

It's impossible to fix it without the sources, so I'll leave it to someone who knows the JDBC driver better to perform this update. 

Closing

**Comment by Sanne** [comment]
Just out of curiousity, what did you do to get the `TCP_KEEPIDLE` ?

**Comment by gastaldi** [comment]
@Sanne I bumped the devservice image to 12.1.0.0 too, do you think it may be related?

**Comment by Sanne** [comment]
I don't know about the devservice.. I couldn't try this out, just curious how you got the `TCP_KEEPIDLE` : is that in native only or affecting JVM mode as well?

**Comment by gastaldi** [comment]
Seems to happen only in native. Logs: https://github.com/quarkusio/quarkus/actions/runs/12037826792/job/33562670423?pr=44752

**Comment by gsmet** [comment]
I wonder if they somehow get this value by reflection cause it's new in Java 21+ and thus we would need to add it.

**Comment by Sanne** [comment]
ah if it's just native then I might have a hunch related to class initialization. I'll try to look into it but no promises

**Comment by Sanne** [comment]
I've found this GralVM issue which looks related and has a possible workaround, but it didn't work for me in this case :
 - https://github.com/oracle/graal/issues/6457#issuecomment-1723413415

Weird, I've asked the Mandrel team for suggestions.

**Pull request closed without merging**